src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Inclusion/InclusionContext.cs

```diff
@@ -89,9 +110,11 @@ public static bool IsCircularReference(object file, out IEnumerable<object> depe
         {
             dependencyChain = null;
 
-            if (t_files.Contains(file))
+            var markupStack = t_markupStacks.Value;
+            var inclusionStack = markupStack.Count > 0 ? markupStack.Peek().inclusionStack : null;

```

Comment:
>var inclusionStack = markupStack.Count > 0 ? markupStack.Peek().inclusionStack : null; [](start = 12, length = 86)

consider A @ B Include A, A is markdown, @B ? displayPropertyName= Title, B is yaml, title property is markdown property which includes A

Reply:
@ will have circular reference, markdown inclusion does not have circular reference, this is expected.

Reply:
but the error message will be confused,

Reply:
we need consider this case, how about create a new issue for tracking(cross type of circle referencing handling), @yufeih , @live1206

Reply:
agreed, we need to verify the case described in this PR firstly